tests/browser_context_test.go

```diff
@@ -189,3 +192,243 @@ func TestBrowserContextAddCookies(t *testing.T) {
+	okHandler := func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}
+
+	tests := map[string]struct {
+		// setupHandler is a handler that will be used to setup the
+		// test environment. it acts like a page returning cookies.
+		setupHandler func(w http.ResponseWriter, r *http.Request)
+
+		// documentCookiesSnippet is a JavaScript snippet that will be
+		// evaluated in the page to set document.cookie.
+		documentCookiesSnippet string
+
+		// wantDocumentCookies is a string representation of the
+		// document.cookie value that is expected to be set.
+		wantDocumentCookies string
+
+		// wantContextCookies is a list of cookies that are expected
+		// to be set in the browser context.
+		wantContextCookies []*api.Cookie
+	}{
+		"no_cookies": {
+			setupHandler: okHandler,
+			documentCookiesSnippet: `
+				() => {
+					return document.cookie;
+				}
+			`,
+			wantDocumentCookies: "",
+			wantContextCookies:  nil,
+		},
+		"cookie": {
+			setupHandler: okHandler,
+			documentCookiesSnippet: `
+				() => {
+					document.cookie = "name=value";
+					return document.cookie;
+				}
+			`,
+			wantDocumentCookies: "name=value",
+			wantContextCookies: []*api.Cookie{
+				{
+					Name:     "name",
+					Value:    "value",
+					Domain:   "127.0.0.1",
+					Expires:  -1,
+					HTTPOnly: false,
+					Path:     "/",
+					SameSite: "",
+					Secure:   false,
+				},
+			},
+		},
+		"cookies": {
+			setupHandler: okHandler,
+			documentCookiesSnippet: `
+				() => {
+					document.cookie = "name=value";
+					document.cookie = "name2=value2";
+					return document.cookie;
+				}
+			`,
+			wantDocumentCookies: "name=value; name2=value2",
+			wantContextCookies: []*api.Cookie{
+				{
+					Name:     "name",
+					Value:    "value",
+					Domain:   "127.0.0.1",
+					Expires:  -1,
+					HTTPOnly: false,
+					Path:     "/",
+					SameSite: "",
+					Secure:   false,
+				},
+				{
+					Name:     "name2",
+					Value:    "value2",
+					Domain:   "127.0.0.1",
+					Expires:  -1,
+					HTTPOnly: false,
+					Path:     "/",
+					SameSite: "",
+					Secure:   false,
+				},
+			},
+		},
+		"cookie_with_path": {
+			setupHandler: okHandler,
+			documentCookiesSnippet: `
+				() => {
+					document.cookie = "name=value; path=/empty";
+					return document.cookie;
+				}
+			`,
+			wantDocumentCookies: "name=value",
+			wantContextCookies: []*api.Cookie{
+				{
+					Name:     "name",
+					Value:    "value",
+					Domain:   "127.0.0.1",
+					Expires:  -1,
+					HTTPOnly: false,
+					Path:     "/empty",
+					SameSite: "",
+					Secure:   false,
+				},
+			},
+		},
+		"cookie_with_different_domain": {
+			setupHandler: okHandler,
+			documentCookiesSnippet: `
+				() => {
+					document.cookie = "name=value; domain=k6.io";
+					return document.cookie;
+				}
+			`,
+			wantDocumentCookies: "", // some cookies cannot be set (i.e. cookies using different domains)
+			wantContextCookies:  nil,
+		},
```

Comment:
This test case seems to test more the browser implementation than ours, but I guess it doesn't harm either.

Reply:
Yeah, it is. I actually quite like it. For someone who isn't clued up on cookies this is useful to see 🙂

Reply:
You're right 👍 I'll remove this.

Reply:
Let's keep it, then :)

Reply:
We can keep it :+1: